app/workers/email/send_user_digest_worker.rb

```diff
@@ -0,0 +1,23 @@
+module Email
+  class SendUserDigestWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low_priority, retry: 15, lock: :until_executing
+
+    def perform(user_id)
+      user = User.find_by(id: user_id)
+      return unless user&.email_digest_periodic? && user&.registered?
+
+      user_email_heuristic = EmailLogic.new(user).analyze
+      return unless user_email_heuristic.should_receive_email?
+
+      articles = user_email_heuristic.articles_to_send
+      begin
+        DigestMailer.with(user: user, articles: articles).digest_email.deliver_now
+      rescue StandardError => e
```

Comment:
I kept this logic from the rake task. We could probably remove it but for now I just ported it over.

Reply:
make sense. We should eventually move all this logic rich workers to services. Haven't reached a solid opinion on that :D